Fix/rebuild db

PGijsbers opened the pull request:
The `rebuild-db` option "no" would still create the database if it did not exist yet. However, when using MySQL group replication the databases for the secondary nodes need to be created by MySQL's replication process, not the REST API.
This PR makes it so that setting `rebuild-db` to "no" does not affect the database in any way. If this is set and the database does not exist, you will get a warning (as this is _only_ for users using group replication, in all other cases it will lead to errors on every interaction).

PGijsbers committed:
Do not setup the database if rebuild-db is set to no

PGijsbers committed:
Log a warning when the database does not exist and is not created

PGijsbers commented:
Closes #299

PGijsbers commented on src/main.py:
```diff
@@ -110,18 +111,27 @@ def create_app() -> FastAPI:
-            add_delete_triggers(AIoDConcept)
+    if args.rebuild_db == "no":
+        if not database_exists():
+            logging.warning(
```
We probably want to just exit with an error here.

Taniya-Das replied:
Yes, looks appropriate to me.

PGijsbers replied:
Okay, I will add a rebuild-db mode that specifies explicitly that you are using group replication, and otherwise in this scenario raise and error.

Taniya-Das commented on src/main.py:
```diff
@@ -110,18 +111,27 @@ def create_app() -> FastAPI:
+                "AI-on-Demand database does not exist on the MySQL server, "
+                "but `rebuild_db` is set to 'no'. If you are not creating the "
+                "database through other means, such as MySQL group replication, "
+                "this likely means that you will errors or undefined behavior."
```
Correction - "this likely means that you will get errors or undefined behaviour."

PGijsbers replied:
```suggestion
                "this likely means that you will get errors or undefined behavior."
```

PGijsbers committed:
Fix typo in warning

PGijsbers committed:
Rename parameter and options for clarity and update outofdate docs

PGijsbers commented:
@Taniya-Das Sorry, I went a different way. I thought it was getting awfully confusing, so instead I decided to rename the parameter to better reflect its default purpose: creating the database. I updated the docs as well, and found some outdated documentation that I also just updated. I hope this will be much clearer for users, WDYT?

Taniya-Das commented on README.md:
```diff
@@ -178,33 +178,22 @@ for a production instance.
+### Populating the Database
+To populate the database with some examples, run the `connectors/fill-examples.sh` script.
+When using `docker compose` you can easily do this by running the "examples" profile:
+`docker compose --profile examples up`
```
I am checking, why you decided to omit the "limit" when populating the database, as it looks useful.

Taniya-Das replied:
The rest looks good, and the options for creating the database are more clear now.

PGijsbers replied:
I didn't, it was already no longer available in the application, but the documentation was out of date.

Taniya-Das replied:
oh oops, you are right.

Taniya-Das reviewed (commented):
Looks good, however I see the huggingface unittest failing.

PGijsbers committed:
Merge branch 'develop' into fix/rebuild-db
